File: tusow/src/main/java/it/unibo/coordination/tusow/api/LogicTupleSpaceApi.java
package it.unibo.coordination.tusow.api;

import alice.tuprolog.Term;
import io.vertx.ext.web.RoutingContext;
import it.unibo.coordination.linda.logic.LogicMatch;
import it.unibo.coordination.linda.logic.LogicTemplate;
import it.unibo.coordination.linda.logic.LogicTuple;

public interface LogicTupleSpaceApi extends TupleSpaceApi<LogicTuple, LogicTemplate, String, Term, LogicMatch> {
    static LogicTupleSpaceApi get(RoutingContext context) {
        return new LogicTupleSpaceApiImpl(context);
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/api/LogicTupleSpaceApi.kt
package it.unibo.coordination.tusow.api

import io.vertx.ext.web.RoutingContext
import it.unibo.coordination.linda.logic.LogicMatch
import it.unibo.coordination.linda.logic.LogicTemplate
import it.unibo.coordination.linda.logic.LogicTuple
import it.unibo.tuprolog.core.Term

interface LogicTupleSpaceApi : TupleSpaceApi<LogicTuple, LogicTemplate, String, Term, LogicMatch> {
    companion object {
        @JvmStatic
        operator fun get(context: RoutingContext): LogicTupleSpaceApi {
            return LogicTupleSpaceApiImpl(context)
        }
    }
}